docs/appendices/release-notes/5.9.0.rst

```diff
@@ -53,6 +53,12 @@ Breaking Changes
 
   It is a breaking change because the fix causes performance degradations.
 
+- Fixed an issue that caused ``WHERE`` clause containing
+  :ref:`scalar-array_length` under ``<``, ``<=`` or ``=`` to return invalid
+  results. It is a breaking change because the fix causes performance
+  degradations but is addressed for the arrays in the tables created on and
+  after 5.9.0.
```

Comment:
reads like fix is done for new tables. Maybe phrase like 
`...because the fix causes performance regressions on tables created on CrateDB<5.9.0`.  
`On new tables fix has positive impact on performance.`